regulation-worker/internal/delete/api/api.go

```diff
@@ -257,3 +303,54 @@ func (api *APIManager) refreshOAuthToken(destination *model.Destination, job mod
 	}
 	return nil
 }
+
+type PostResponseParams struct {
+	destination              model.Destination
+	isOAuthEnabled           bool
+	currentOAuthRetryAttempt int
+	job                      model.Job
+	oAuthDetail              oauthDetail
+	responseBodyBytes        []byte
+	responseStatusCode       int
+}
+
+func (api *APIManager) PostResponse(ctx context.Context, params PostResponseParams) model.JobStatus {
```

Comment:
It seems like there are two different receiver names being used. One as `m` and other as `api`. Can we use one name instead like say `m`?
```suggestion
func (m *APIManager) PostResponse(ctx context.Context, params PostResponseParams) model.JobStatus {
```